integration-test/http/http_system_test.c

```diff
@@ -0,0 +1,350 @@
+                ( int32_t ) SERVER_HOST_LENGTH, SERVER_HOST,
+                ( int32_t ) requestInfo.pathLen, requestInfo.pPath ) );
+    LogDebug( ( "Request Headers:\n%.*s\n"
+                "Request Body:\n%.*s\n",
+                ( int32_t ) requestHeaders.headersLen,
+                ( char * ) requestHeaders.pBuffer,
+                ( int32_t ) REQUEST_BODY_LENGTH, REQUEST_BODY ) );
+
+    /* Send the request and receive the response. */
+    TEST_ASSERT_EQUAL( HTTP_SUCCESS, HTTPClient_Send( pTransportInterface,
+                                                      &requestHeaders,
+                                                      ( uint8_t * ) REQUEST_BODY,
+                                                      REQUEST_BODY_LENGTH,
+                                                      &response,
+                                                      0 ) );
+
+    LogDebug( ( "Received HTTP response from %.*s%.*s...\n"
+                "Response Headers:\n%.*s\n"
+                "Response Status:\n%u\n"
+                "Response Body:\n%.*s\n",
+                ( int32_t ) SERVER_HOST_LENGTH, SERVER_HOST,
+                ( int32_t ) requestInfo.pathLen, requestInfo.pPath,
+                ( int32_t ) response.headersLen, response.pHeaders,
+                response.statusCode,
+                ( int32_t ) response.bodyLen, response.pBody ) );
+
+    /* Close connection if a "Connection: close" header is found */
+    if( response.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
+    {
+        ( void ) Openssl_Disconnect( &networkContext );
+    }
+
+    /* Verify that content length is greater than 0 for GET requests. */
+    if( strcmp( pMethod, HTTP_METHOD_GET ) )
+    {
+        TEST_ASSERT_GREATER_THAN( 0, response.contentLength );
+    }
+
+    /* Verify response body is present */
+    if( strcmp( pMethod, HTTP_METHOD_HEAD ) )
+    {
+        TEST_ASSERT_GREATER_THAN( 0, response.bodyLen );
+    }
+}
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+    /* Establish TLS session with HTTP server on top of a newly-created TCP connection. */
+    connectToServerWithBackoffRetries( &networkContext );
+
+    /* Define the transport interface. */
+    ( void ) memset( &transportInterface, 0, sizeof( transportInterface ) );
+    transportInterface.recv = Openssl_Recv;
+    transportInterface.send = Openssl_Send;
+    transportInterface.pNetworkContext = &networkContext;
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+    /* End TLS session, then close TCP connection. */
+    ( void ) Openssl_Disconnect( &networkContext );
+}
+
+/* ========================== Test Cases ============================ */
+
+/**
+ * @brief Sends a GET request synchronously and verifies the result.
+ */
+void test_HTTP_GET_Request( void )
+{
+    sendHttpRequest( &transportInterface,
```

Comment:
For stabilization of the tests, I think we should also retry failed requests with exponential back off.